text/0000-ts-adoption-plan.md

```diff
@@ -0,0 +1,780 @@
+    - [Ember Classic features](#ember-classic-features)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Documentation](#documentation)
+  - [Publishing types](#publishing-types)
+    - [Ambient types vs. types-from source](#ambient-types-vs-types-from-source)
+    - [Classic Features](#classic-features-1)
+  - [SemVer](#semver)
+  - [Documentation](#documentation-1)
+- [Unresolved questions](#unresolved-questions)
+
+
+## Motivation
+
+The overall motivation for this work is the same as that in [RFC #0724][rfc-0724]:
+
+> In sum, making TypeScript an officially supported language for Ember will benefit all Ember users, JavaScript and TypeScript alike; it will solve many pain points for TypeScript users that cannot otherwise be addressed; and it will close a gap for Ember compared to other frameworks.
+
+Where that RFC was concerned with *whether* we should pursue those goals, this RFC is concerned with *how* we accomplish them. The specific approach described here aims to roll out TypeScript support in a way that:
+
+- maintains Ember’s strong stability guarantees
+- allows us to make steady incremental progress
+- communicates progress clearly to the community
+- minimizes the migration costs for existing Ember TypeScript users
+
+
+## Detailed design
+
+To fully support TypeScript across the Ember ecosystem, we need:
+
+- a Semantic Versioning policy which can absorb breaking changes in TypeScript itself *without* imposing breaking changes on Ember developers
+- an edition support policy which tells us what our types *must* support (as distinct from what they *may* support)
+- a plan for migrating existing Ember TypeScript users from DefinitelyTyped to using Ember’s own core types
+- a plan for CLI integration (e.g. `ember new --typescript`)
+- test infrastructure for Ember’s types, both to prevent regressions and to catch breaking changes from TypeScript early so they can be mitigated
+- release infrastructure to allow us to handle pre-release testing, feature flags, and alpha and beta releases
+- updates to our documentation to include TypeScript as a first-class citizen of the ecosystem
+
+
+### Semantic Versioning
+
+Ember packages which publish types will adhere to the [Semantic Versioning for TypeScript Types][rfc-0730] proposal. See that RFC for details on strategies for mitigating breaking changes from TypeScript, and see [**How We Teach This: Documenting SemVer**](#documenting-semver) below for discussion of how we will document this at the framework level.
+
+The 10,000-foot summary of our treatment of SemVer for TypeScript is:
+
+> Code which type checks at any given point in a major release of Ember packages will continue to type check throughout that Ember major release for consumers who use supported TypeScript versions.
+
+This is **the “no new red squiggles rule”:** if you use supported TypeScript versions, you will not see new “red squiggles” for type errors in your editor in an Ember major version.
```

Comment:
```suggestion
This is **the “no new red squiggles rule”:** if you use supported TypeScript versions, you will not see new “red squiggles” for type errors in your editor for Ember minor version upgrades.
```

(I don't love this but haven’t come up with something I *do* love. Input welcome.)